### PR TITLE
Issue/7574 stock photo featured image

### LINKED ISSUE #7574: Stock Photos: support featured image
The new "stock photos" feature isn't available when choosing a featured image. We should add that, which will require adding the ability to choose a single stock photo (ie: no multiselect).

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #7574 - adds stock photos as a source for choosing a featured image. Demo GIF below, high-res video [here](https://cloudup.com/cm272-p4gol).

stock-featured

### CODE REVIEW

**Comment by iamthomasbishop** [comment]
Looking great, @nbradbury! Couple notes:

* Can we use the `image-multiple` icon from Gridicons, instead of `image`?
* Can we bring the icon size down slightly to match the ones on the other Media Picker toolbar? I believe these are 24x24?

Other than that, it looks great functionally. There's definitely room to improve the flow itself in the future, but for the time being this is a nice quick win.

**Comment by nbradbury** [comment]
@iamthomasbishop In ab1cc90 I updated the icons to match those in the media picker (editor).

screenshot_1522700681

**Comment by iamthomasbishop** [comment]
Just gave the apk a spin. Functionally perfect. Caught one tiny design detail that I didn't catch at first glance - the border is _super_ dim on the toolbar. Can we match the editor's formatting toolbar border-top? It looks like $grey-lighten-10 but I'm not sure if it's that or a color with semi-transparency. Here's what it'd look like (current vs. proposed):

screen shot 2018-04-02 at 4 54 54 pm

**Comment by nbradbury** [comment]
> Caught one tiny design detail that I didn't catch at first glance - the border is super dim on the toolbar. Can we match the editor's formatting toolbar border-top (1pt)?

Added in 788fd66.

screenshot_1522708183

**Comment by iamthomasbishop** [comment]
LGTM! 🚢 it!

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerActivity.java [hunk 1]
```diff
@@ -243,6 +251,13 @@ private void launchWPMediaLibrary() {
         }
     }
 
+    private void launchStockMediaPicker() {
+        if (mSite != null) {
+            ActivityLauncher.showStockMediaPickerForResult(this,
+                    mSite, RequestCodes.STOCK_MEDIA_PICKER_SINGLE_SELECT);
+        }
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerActivity.java]: can we add a Toast if `mSite` is null? I know such a situation is unlikely to happen, but in that event then no action is performed and the user is not informed. Maybe we can fail in the `onCreate` method in the Activity itself. Not a direct part of this PR but it seems there are several checks on `mSite` not being null that could benefit from having this check right from start.

**Comment by mzorz** [comment]
This is an awesome feature @nbradbury, it's nice to see it come to this corner of the app as well! 

Some things that I found while testing:

1. the empty view (that reads “Search to find free photos…” is not keep on rotation

2. if you start an upload and rotate the screen, the upload doesn’t end properly, furthermore the ProgressDialog leaks as shown here:

```
04-09 21:42:15.233 18126-19846/org.wordpress.android V/FA: Activity paused, time: 2788191
04-09 21:42:15.235 18126-18237/org.wordpress.android D/OpenGLRenderer: endAllStagingAnimators on 0x98cf4e80 (RippleDrawable) with handle 0x99fc9810
04-09 21:42:15.244 18126-18126/org.wordpress.android E/WindowManager: android.view.WindowLeaked: Activity org.wordpress.android.ui.stockmedia.StockMediaPickerActivity has leaked window com.android.internal.policy.impl.PhoneWindow$DecorView{231f65b7 V.E..... R......D 0,0-228,88} that was originally added here
                                                                          at android.view.ViewRootImpl.<init>(ViewRootImpl.java:363)
                                                                          at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:261)
                                                                          at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:69)
                                                                          at android.app.Dialog.show(Dialog.java:298)
                                                                          at org.wordpress.android.ui.stockmedia.StockMediaPickerActivity.showUploadProgressDialog(StockMediaPickerActivity.java:327)
                                                                          at org.wordpress.android.ui.stockmedia.StockMediaPickerActivity.uploadSelection(StockMediaPickerActivity.java:522)
                                                                          at org.wordpress.android.ui.stockmedia.StockMediaPickerActivity.access$100(StockMediaPickerActivity.java:61)
                                                                          at org.wordpress.android.ui.stockmedia.StockMediaPickerActivity$1.onClick(StockMediaPickerActivity.java:170)
                                                                          at android.view.View.performClick(View.java:4756)
                                                                          at android.view.View$PerformClick.run(View.java:19749)
                                                                          at android.os.Handler.handleCallback(Handler.java:739)
                                                                          at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                          at android.os.Looper.loop(Looper.java:135)
                                                                          at android.app.ActivityThread.main(ActivityThread.java:5221)
                                                                          at java.lang.reflect.Method.invoke(Native Method)
                                                                          at java.lang.reflect.Method.invoke(Method.java:372)
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
04-09 21:42:15.273 18126-18126/org.wordpress.android V/FA: onActivityCreated
```

Both issues can be seen in this animation:
featuredimg


We should probably use a `DialogFragment` to avoid losing state here (in `StockMediaPickerActivity`), and then saving in `onSaveInstanceState` the media being uploaded, then checking the current uploads FluxC MediaStore and check if whatever we saved in our temporal class member in onSaveInstanceState exists. Then if we were processing but didn't  "hear" the `OnStockMediaUploaded` event (because we destroyed the activity thus weren't registered to FluxC's dispatcher), properly run the code in that method so things continue just the way they were meant to. 
I think this could work, but let me know if it's not super clear

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerActivity.java (hunk 1 above)
Comment (nbradbury) [reply WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerActivity.java]: Added in cbbbb75, along with ensuring the stock photo menu item doesn't appear when there isn't a site.

**Comment by nbradbury** [comment]
@mzorz Thanks for the review, and for catching the rotation issues. Is it okay with you if I tackle those in a separate PR, since those problems existed prior to this?

**Comment by mzorz** [comment]
>  Is it okay with you if I tackle those in a separate PR, since those problems existed prior to this?

Sure thing! :shipit:

**Pull request merged**